### ISSUE TITLE
Question about parsing EG_LOG_LEVEL to int

### ISSUE THREAD

**Issue by taeyeopkim1** [jupyter/enterprise_gateway#962]
## Description

I have a question while using DockerProcessProxy.

I ran EG as below.
```
docker run -d --name test-eg elyra/enterprise-gateway:2.1.1
docker exec -it test-eg bash
```

I modified "kernel.json" in "/usr/local/share/jupyter/kernels/python_docker" as shown below.

[python_docker/kernel.json](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernelspecs/python_docker/kernel.json)
```
"class_name": "enterprise_gateway.services.processproxies.docker_swarm.DockerProcessProxy"
"image_name": "elyra/kernel-py:2.1.1"
```

I also modified "launch_docker.py" in "/usr/local/share/jupyter/kernels/python_docker/scripts" by referring to [here](https://github.com/jupyter/enterprise_gateway/commit/d0d4c741cfe2c88b6790f7be3bcd5537778b01b5#diff-48788ad7ad38747981e888a233dd9f3f7b1e0cc3ca613c74296e6ee2542635b6).
```
kwargs['group_add'] = [group, '100']
```

I ran the kernel using the api as shown below.
```
curl -d '{"name":"python_docker"}' -X POST -H "Content-Type: application/json" http://localhost:8888/api/kernels
```

When I print the log [here](https://github.com/jupyter/enterprise_gateway/blob/v2.1.1/etc/kernel-launchers/docker/scripts/launch_docker.py#L88), EG_LOG_LEVEL is set to DEBUG.
This is because the default value is DEBUG in [start-enterprise-gateway.sh](https://github.com/jupyter/enterprise_gateway/blob/v2.1.1/etc/docker/enterprise-gateway/start-enterprise-gateway.sh#L24)

However, an error has occurred while converting DEBUG to int at [launch_ipykernel.py](https://github.com/jupyter/enterprise_gateway/blob/v2.1.1/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L22)
```
log_level = int(os.getenv('EG_LOG_LEVEL', '10'))
```
Error log
```
Traceback (most recent call last):
  File "/usr/local/bin/kernel-launchers/python/scripts/launch_ipykernel.py", line 22, in <module>
    log_level = int(os.getenv('EG_LOG_LEVEL', '10'))
ValueError: invalid literal for int() with base 10: 'DEBUG'
```

#### To bypass
- Add EG_LOG_LEVEL environment when I run eg.
```
docker run -d -e EG_LOG_LEVEL=10 --name test-eg elyra/enterprise-gateway:2.1.1
```
OR
-  Modify "[launch_docker.py](https://github.com/jupyter/enterprise_gateway/blob/v2.1.1/etc/kernel-launchers/docker/scripts/launch_docker.py#L44)"
```
    # Since the environment is specific to the kernel (per env stanza of kernelspec, KERNEL_ and ENV_WHITELIST)
    # just add the env here.
    param_env.update(os.environ)
    param_env.pop('PATH')  # Let the image PATH be used.  Since this is relative to images, we're probably safe.
    param_env['EG_LOG_LEVEL'] = '10'
```

---
So, why does [launch_ipykernel.py](https://github.com/jupyter/enterprise_gateway/blob/v2.1.1/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L22) convert EG_LOG_LEVEL to int?
```
log_level = int(os.getenv('EG_LOG_LEVEL', '10'))
```
(I also see it in the [master](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L26) branch.)

---

## Screenshots / Logs
```
Traceback (most recent call last):
  File "/usr/local/bin/kernel-launchers/python/scripts/launch_ipykernel.py", line 22, in <module>
    log_level = int(os.getenv('EG_LOG_LEVEL', '10'))
ValueError: invalid literal for int() with base 10: 'DEBUG'
```

## Environment

- Enterprise Gateway Version [2.1.1, master]
- Platform: [Docker, docker_swarm.DockerProcessProxy]
- Others [Docker Hub: elyra/enterprise-gateway:2.1.1, elyra/kernel-py:2.1.1]

**Comment by kevin-bates** [comment]
Hi @taeyeopkim1 - thanks for taking a look at DockerProcessProxy - it probably hasn't had a lot of use.

Good catch on the launcher script.  I guess only docker transfers the env from EG to the kernel containers and why we haven't seen this.  I think the correct change is to convert: `log_level = int(os.getenv('EG_LOG_LEVEL', '10'))` to `log_level = os.getenv('EG_LOG_LEVEL', 10)`.  This will still enable the flow of the string (`'DEBUG'`) approach.

Based on your comments, it looks like other changes may be on the way relative to Docker support.  Are you planning on contributing these changes?  If so, please include this update - otherwise, we'll make this change in a separate PR.

Sorry for the inconvenience.

**Comment by taeyeopkim1** [comment]
Thanks a lot for your reply.
I checked the code where "log_level" was used.

[python/scripts/launch_ipykernel.py](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L26)
[R/scripts/gateway_listener.py](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/R/scripts/gateway_listener.py#L20)
```
log_level = int(os.getenv('EG_LOG_LEVEL', '10'))
```
[launch_ipykernel.py#L31](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L31)
[gateway_listener.py#L25](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/R/scripts/gateway_listener.py#L25)
```
logger.setLevel(log_level)
```

https://github.com/python/cpython/blob/3.6/Lib/logging/__init__.py#L823-L827
```
    def setLevel(self, level):
        """
        Set the logging level of this logger.  level must be an int or a str.
        """
        self.level = _checkLevel(level)
```

https://github.com/python/cpython/blob/3.6/Lib/logging/__init__.py#L190-L199
```
def _checkLevel(level):
    if isinstance(level, int):
        rv = level
    elif str(level) == level:
        if level not in _nameToLevel:
            raise ValueError("Unknown level: %r" % level)
        rv = _nameToLevel[level]
    else:
        raise TypeError("Level not an integer or a valid string: %r" % level)
    return rv
```

https://github.com/python/cpython/blob/3.6/Lib/logging/__init__.py#L109-L118
```
_nameToLevel = {
    'CRITICAL': CRITICAL,
    'FATAL': FATAL,
    'ERROR': ERROR,
    'WARN': WARNING,
    'WARNING': WARNING,
    'INFO': INFO,
    'DEBUG': DEBUG,
    'NOTSET': NOTSET,
}
```

In conclusion, as you mentioned, 
I can provide a pull request to convert: `log_level = int(os.getenv('EG_LOG_LEVEL', '10'))` to `log_level = os.getenv('EG_LOG_LEVEL', 10)` in "python/scripts/launch_ipykernel.py" and "R/scripts/gateway_listener.py".

**Comment by taeyeopkim1** [comment]
I modified the code.
```
log_level = os.getenv('EG_LOG_LEVEL', '10')
log_level = int(log_level) if log_level.isdigit() else log_level
```

The code below can cause problems in the following situations:
```
log_level = os.getenv('EG_LOG_LEVEL', 10)
```

```
export EG_LOG_LEVEL=10

>>> import os
>>> log_level = os.getenv('EG_LOG_LEVEL', 10)
>>> type(log_level)
<class 'str'>
>>> log_level
'10'

>>> import logging
>>> log_level = '10'
>>> logger = logging.getLogger('test')
>>> logger.setLevel(log_level)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Users\terre\AppData\Local\Programs\Python\Python38\lib\logging\__init__.py", line 1409, in setLevel
     self.level = _checkLevel(level)
   File "C:\Users\terre\AppData\Local\Programs\Python\Python38\lib\logging\__init__.py", line 194, in _checkLevel
     raise ValueError("Unknown level: %r" % level)
 ValueError: Unknown level: '10'

>>> log_level = 10
>>> logger.setLevel(log_level) # no error
```
